libraries/ts/src/margin/marginAccount.ts

```diff
@@ -613,14 +613,22 @@ export class MarginAccount {
     await this.registerPosition(tokenMintAddress)
     await this.refresh()
 
+    throw new Error("Unable to register position.")
+  }
+
+  async withRegisterPositionIfNotExists(tokenMint: Address, ixArray: TransactionInstruction[]) {
+    assert(this.info)
+    const tokenMintAddress = translateAddress(tokenMint)
+
     for (let i = 0; i < MAX_POSITIONS; i++) {
       const position = this.info.positions.positions[i]
       if (position.token.equals(tokenMintAddress)) {
-        return position
+        return
       }
     }
 
-    throw new Error("Unable to register position.")
+    await this.withRegisterPosition(ixArray, tokenMint)
+    await this.refresh()
```

Comment:
There's no point to refreshing because the account only changes after the instruction is executed
```suggestion
    await this.withRegisterPosition(ixArray, tokenMint)
```